sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/BeamJoinTransforms.java

```diff
@@ -116,6 +116,7 @@ public JoinAsLookup(
       this.outputSchema = outputSchema;
       this.factColOffset = factColOffset;
       joinFieldsMapping(joinCondition, factColOffset, lkpColOffset);
+      this.seekableTable.setJoinSubsetType(joinSubsetType);
```

Comment:
Will this work? at this point `JoinAsLookup.joinSubsetType` is used before assigned

Reply:
joinSubsetType is assigned in joinFieldsMapping, and after that I invoke seekableTable.setJoinSubsetType

Reply:
maybe we can assigning it when invoke setup?

Reply:
missed that joinFieldsMapping is setting up multiple class members. sorry for confusion.